### DISCUSSION TITLE
Selecting profiles poligonally or bathymetry-based

### DISCUSSION THREAD

**Discussion by ElenaTerzic** [euroargodev/argopy#368] [General]
Hi!

I wondered if it's possible to select a region which is not rectangular, but polygonal, see figure attached:
![Otranto](https://github.com/euroargodev/argopy/assets/57098850/d44a77c9-1288-4461-876e-021a3a71a0cb)

Maybe the solution could be to define it with multiple rectangles and somehow concatenate these data sets?

I would like to select profiles based on the basin bathymetry. For example, looking at Argo data only for locations that are shallower than a certain depth (say 1000 m).

Is there a way to do this on Argopy?

Thanks in advance!

**Comment by gmaze** [comment]
Hi @ElenaTerzic 
this is something we didn't get the time to implement yet
in your use-case, you can 1st get the data for the rectangular domain, then [fetch bathymetry with argopy as well](https://argopy.readthedocs.io/en/latest/user-guide/working-with-argo-data/data_quality_control.html#topography) and then mask your dataset easily
g
